### PR TITLE
deepin.dde-kwin: init at 5.0.0

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Add dde-kwin https://github.com/linuxdeepin/dde-kwin.

Other than packaging this, we need to update deepin to its latest revisions.
Which appears to be a bit of a difficult task, because I cannot find where recent changes are documented. i.e I think some other things were deprecated as well. Probably just have to attempt to update and see what breaks.

###### Things done
It builds, I don't think any other testing can be done until deepin packages are updated.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by romildo** [comment]
Building locally is failing due to a dependence (`qt5dxcb-plugin`) that fails to build:

```
[...]
cp: missing destination file operand after 'platformplugin/libqt5xcbqpa-dev/5.12.5/'
Try 'cp --help' for more information.
note: keeping build directory '/tmp/nix-build-qt5dxcb-plugin-5.0.1.drv-1'
builder for '/nix/store/76m40p6jdi9bi89p7lnyrarwdjjmiaiv-qt5dxcb-plugin-5.0.1.drv' failed with exit code 1
cannot build derivation '/nix/store/aapb9g8s7309h1pky72myssgi0aly7g2-qt5integration-0.3.12.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/mf5k4njhg4a316g37fkkki491pgvn8jl-dde-kwin-5.0.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/mf5k4njhg4a316g37fkkki491pgvn8jl-dde-kwin-5.0.0.drv' failed
```

**Comment by worldofpeace** [comment]
@romildo You must have rebase this PR onto a older revision, that's fixed in https://github.com/NixOS/nixpkgs/commit/c2eed8b9b08b0c435b6674bf052433de946fefe4. ofborg seems to have succeeded in building it.

**Comment by romildo** [comment]
The installed file `$out/lib/pkgconfig/dde-kwin.pc` contains the line below where the prefix is repeated:
```
libdir=/nix/store/zgfgfak1sp8ahc87r9qs9qmwgd65yiqf-dde-kwin-5.0.0//nix/store/zgfgfak1sp8ahc87r9qs9qmwgd65yiqf-dde-kwin-5.0.0/lib
```

File: pkgs/desktops/deepin/dde-kwin/default.nix [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  postFixup = ''
+    chmod +x $out/bin/kwin_no_scale
+  '';
+
+  enableParallelBuilding = true;
+
+  passthru.updateScript = deepin.updateScript { name = "${pname}-${version}"; };
+
+  meta = with stdenv.lib; {
+    description = "KWin configures on DDE";
```
Comment (romildo) [inline pkgs/desktops/deepin/dde-kwin/default.nix]: Maybe _KWin configuration for Deepin Desktop Environment_ is a better description. The upstream one seems grammatically wrong to me.

Comment (worldofpeace) [reply pkgs/desktops/deepin/dde-kwin/default.nix]: Indeed, it is grammatically incorrect. I think I can open an issue as well to change it to this.

**Comment by worldofpeace** [comment]
@romildo Do you have any idea what `-L/usr/X11R6/lib64` is supposed to be?
> perhaps the X11R6 is a debianism?

File: pkgs/desktops/deepin/dde-kwin/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/desktops/deepin/dde-kwin/default.nix]: Ah nvm, they don't have issues enabled on the repo...

**Comment by romildo** [comment]
> @romildo Do you have any idea what `-L/usr/X11R6/lib64` is supposed to be?
> 
> > perhaps the X11R6 is a debianism?

Probably it refers to the location where X11 libraries used to be installed some years ago. As far as I known it is not used anymore and can be ignored. It does not exist in none of my linux installations, including Deepin Linux, Ubuntu, Archlinux, and Fedora.

See the section **The /usr/X11R6 directory** in https://wiki.debian.org/Xorg69To7

**Comment by worldofpeace** [comment]
> > @romildo Do you have any idea what `-L/usr/X11R6/lib64` is supposed to be?
> > > perhaps the X11R6 is a debianism?
> 
> Probably it refers to the location where X11 libraries used to be installed some years ago. As far as I known it is not used anymore and can be ignored. It does not exist in none of my linux installations, including Deepin Linux, Ubuntu, Archlinux, and Fedora.
> 
> See the section **The /usr/X11R6 directory** in https://wiki.debian.org/Xorg69To7

I've removed it from the file at install. Also opened a PR to correct the .pc https://github.com/linuxdeepin/dde-kwin/pull/105 further.

**Review by romildo — APPROVED** [review]
LGTM

**Pull request merged**